src/bun.js/api/bun/socket.zig

```diff
@@ -1880,6 +1954,104 @@ fn NewSocket(comptime ssl: bool) type {
+            callframe: *JSC.CallFrame,
+        ) callconv(.C) JSValue {
+            if (comptime ssl == false) {
+                return JSValue.jsUndefined();
+            }
+
+            if (this.detached) {
+                return JSValue.jsUndefined();
+            }
+
+            const args = callframe.arguments(1);
+
+            if (args.len < 1) {
+                globalObject.throw("Expected session to be a string, Buffer or TypedArray", .{});
+                return .zero;
+            }
+
+            const session_arg = args.ptr[0];
+            var arena: bun.ArenaAllocator = bun.ArenaAllocator.init(bun.default_allocator);
+            defer arena.deinit();
+
+            var exception_ref = [_]JSC.C.JSValueRef{null};
+            var exception: JSC.C.ExceptionRef = &exception_ref;
+            if (JSC.Node.StringOrBuffer.fromJS(globalObject, arena.allocator(), session_arg, exception)) |sb| {
+                var session_slice = sb.slice();
+                var ssl_ptr: *BoringSSL.SSL = @ptrCast(*BoringSSL.SSL, this.socket.getNativeHandle());
+                var tmp = @ptrCast([*c]const u8, session_slice.ptr);
+                const session = BoringSSL.d2i_SSL_SESSION(null, &tmp, @intCast(c_long, session_slice.len)) orelse return JSValue.jsUndefined();
+                if (BoringSSL.SSL_set_session(ssl_ptr, session) != 1) {
+                    globalObject.throwValue(getSSLException(globalObject, "SSL_set_session error"));
+                    return .zero;
+                }
+                return JSValue.jsUndefined();
+            } else if (exception.* != null) {
+                globalObject.throwValue(JSC.JSValue.c(exception.*));
+                return .zero;
+            } else {
+                globalObject.throw("Expected session to be a string, Buffer or TypedArray", .{});
+                return .zero;
+            }
+        }
+
+        pub fn getSession(
+            this: *This,
+            globalObject: *JSC.JSGlobalObject,
+            _: *JSC.CallFrame,
+        ) callconv(.C) JSValue {
+            if (comptime ssl == false) {
+                return JSValue.jsUndefined();
+            }
+
+            if (this.detached) {
+                return JSValue.jsUndefined();
+            }
+
+            var ssl_ptr: *BoringSSL.SSL = @ptrCast(*BoringSSL.SSL, this.socket.getNativeHandle());
+            const session = BoringSSL.SSL_get_session(ssl_ptr) orelse return JSValue.jsUndefined();
+            const size = BoringSSL.i2d_SSL_SESSION(session, null);
+            if (size <= 0) {
+                return JSValue.jsUndefined();
+            }
+            var buffer = bun.default_allocator.alloc(u8, @intCast(usize, size)) catch unreachable;
+            defer bun.default_allocator.free(buffer);
+            var tmp = @ptrCast([*c]u8, buffer.ptr);
+            _ = BoringSSL.i2d_SSL_SESSION(session, &tmp);
+            const result = JSC.ArrayBuffer.createBuffer(globalObject, buffer);
```

Comment:
we could skip this extra allocation, just create the buffer immediately, call `.asArrayBuffer(globalObject).?` on the vale, then write to the array buffer instead of the clone

Reply:
fix on https://github.com/oven-sh/bun/commit/927f54269bdf489d21be4a8a6b89b448fdf48184